### DISCUSSION TITLE
ANT Design not working with Next js 15  due to react js 19 version

### DISCUSSION THREAD

**Discussion by Rasheedullah** [ant-design/ant-design#51558] [General]
Accessing element.ref was removed in React 19. ref is now a regular prop. It will be removed from the JSX Element type in a future release.

i just install Next JS 15, and just try to show ANT Design button but getting error.

**Comment by petercat-assistant** [comment]
The issue you're encountering is due to changes in React 19, where accessing `element.ref` has been removed, and `ref` is now treated as a regular prop. This change can cause compatibility issues with libraries that rely on the previous behavior, such as Ant Design.

To address this issue, you can try the following steps:

1. **Check for Updates**: Ensure that you are using the latest version of Ant Design. The library may have already addressed this compatibility issue in a newer release.

2. **Modify Your Code**: If Ant Design has not yet released a fix, you may need to modify your code to accommodate the changes in React 19. Specifically, you should avoid directly accessing `element.ref` and instead use the `ref` prop as intended in React 19.

3. **Report the Issue**: If the problem persists and there is no available fix, consider reporting the issue to the Ant Design team. They may need to update their library to be compatible with React 19.

4. **Downgrade React**: As a temporary workaround, you could consider downgrading to an earlier version of React that is compatible with Ant Design until a fix is available.

If you need further assistance or specific code modifications, please provide more details about your implementation, and I'll be happy to help! 

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.

**Comment by Wxh16144** [comment]
Related issues https://github.com/ant-design/ant-design/issues/51339

**Reply by Rasheedullah** [reply]
yes, i am also still waiting for solution.

**Comment by hanaessam** [comment]
i'm having the same issue, are there any updates?

**Comment by GurovDmitriy** [comment]
and that's not the only problem

  "dependencies": {
    "@ant-design/nextjs-registry": "^1.0.2",
    "antd": "^5.22.3",
    "next": "15.0.3",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  
  or
  
  "dependencies": {
    "@ant-design/nextjs-registry": "^1.0.2",
    "antd": "^5.22.3",
    "next": "15.0.3",
    "react": "^19.0.0-rc-fb9a90fa48-20240614",
    "react-dom": "^19.0.0-rc-fb9a90fa48-20240614"
  },
  
  click on button...
  
  
![image](https://github.com/user-attachments/assets/741613be-56de-44cf-bc5d-93d43bdc89dc)

**Comment by GurovDmitriy** [comment]
this combination works

  "dependencies": {
    "@ant-design/nextjs-registry": "^1.0.2",
    "antd": "^5.22.3",
    "next": "14.2.19",
    "react": "^18",
    "react-dom": "^18"
  },

**Reply by rasheedullahsmrthb** [reply]
yes with next 14 its work because next js 14 work with react 18, but with next js 15 not working because next js 15 woking with react js 19.

**Comment by anishFE** [comment]
Still waiting for the solution

**Comment by milogus** [comment]
Works for me:

1. Install [Nextjs-registry](https://ant.design/docs/react/use-with-next#using-app-router)
2. Install [Compatibility Package](https://ant-design.antgroup.com/docs/react/v5-for-19?theme=dark#compatibility-package)
3. Disable `--turbopack` from package.json

My scripts and dependencies:

> "scripts": {
    "dev": "next dev -p 80",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@ant-design/nextjs-registry": "^1.0.2",
    "@ant-design/v5-patch-for-react-19": "^1.0.3",
    "antd": "^5.24.4",
    "next": "15.2.3",
    "react": "^19.0.0",
    "react-dom": "^19.0.0"
  }

**Reply by junik1337** [reply]
Thanks!!

**Comment by gcb** [comment]
This issue should be more proeminent on the main site!

**Comment by rizkyhaksono** [comment]
i'm having same issues, any updates?

**Comment by muhammedshamil8** [comment]
Same issue here on Next.js 15 + React 19. AntD breaks with the `element.ref` error.  
Is React 19 support planned for v6, or should we downgrade to React 18 for now?
